### ISSUE TITLE
Students can invite themselves to groups

### ISSUE THREAD

**Issue by mslatour** [PerceptumNL/ReadMore#18]
- when a group is created, a code is generated using `main.models.gen_pass` but then inside save() of group model
- check if code exists, else generate again
- code should be visible in group page by teacher
- students should be able to create their own account using signup together with a field for the group code (optional). 
- if a student is not already in a group, he can enter a group code on the profile page

**Comment by RCJurriaans** [comment]
in readmore/main there is a file adapter.py which extends the defaultaccountadapter found in sitepackages/allauth/account/adapter.py

This file replaces is_open_for_signup(s,r) and more importantly: save_user(s,r,u,f,c)
by extending the signup form (or simply hard coding extra group field in form) you should be able to catch the signup form here.

note that settings.py needs to be updated aswell (see settings.py.example, SIGNUP_ENABLED = True)

**Comment by RCJurriaans** [comment]
```
    if commit:
        # Ability not to commit makes it easier to derive from
        # this adapter by adding
        user.save()
```

after save you can just add the relevant group to the user if a groupcode was detected (or teacher-code for automatic group creation if magic code available)

**Comment by mslatour** [comment]
Generating code for new groups is fixed by f1648a1b9d19edeee79215784bcd3c6f45cfc262 _(code is separated by dashes)_
Automatic adding of users to group by signup code is fixed by: 3bb21d4e6cd3e144362e2c69394349b9fd43c94f and 6e23f2dedab505d86d981dd3c6df5240cbb67c26

**Comment by mslatour** [comment]
Group code is visible to teacher, fixed by d04b7c8

**Comment by mslatour** [comment]
Students can enter group code in profile to join group, fixed by c37038a
